packages/next/server/next-server.ts

```diff
@@ -905,13 +907,36 @@ export default class NextNodeServer extends BaseServer {
         }
 
         if (result.response.headers.has('x-middleware-rewrite')) {
+          const rewritePath = result.response.headers.get(
```

Comment:
Nitpick but I think the destination in the header will always be a full URL, then at some point we make it relative to make a distinction between internal and external rewrites.

Reply:
changed, please take a look again 👯